pkg/agent/controller/trafficcontrol/controller.go

```diff
@@ -0,0 +1,496 @@
+			}
+		} else if version == 2 {
+			externalIDs["erspan_dir"] = dir
+			externalIDs["erspan_hwid"] = hardwareID
+		}
+		portUUID, err := c.ovsBridgeClient.CreateTunnelPortExt(device.Name,
+			ovsconfig.ERSpanTunnel, 0, false, "", remoteIP, "", externalIDs)
+		if err != nil {
+			return 0, err
+		}
+		ofPortTmp, err = c.ovsBridgeClient.GetOFPort(device.Name, false)
+		if err != nil {
+			return 0, err
+		}
+		itf := interfacestore.NewTunnelInterface(device.Name, ovsconfig.ERSpanTunnel, nil, false)
+		itf.OVSPortConfig = &interfacestore.OVSPortConfig{PortUUID: portUUID, OFPort: ofPortTmp}
+		c.interfaceStore.AddInterface(itf)
+		ofPort = uint32(ofPortTmp)
+
+		klog.InfoS("Create ERSPAN", "tunnelID", tunnelID, "remoteIP", remoteIP, "version", version, "index", index,
+			"dir", dir, "hardwareID", hardwareID)
+	}
+	return
+}
+
+func (c *Controller) listOVSPorts() error {
+	ovsPorts, err := c.ovsBridgeClient.GetPortList()
+	if err != nil {
+		return err
+	}
+	ports := make(map[string]ovsconfig.OVSPortData, len(ovsPorts))
+	for i := range ovsPorts {
+		ports[ovsPorts[i].Name] = ovsPorts[i]
+	}
+	c.ovsPorts = ports
+	return nil
+}
+
+func (c *Controller) syncTrafficControl(tcName string) (err error) {
+	startTime := time.Now()
+	defer func() {
+		klog.V(4).InfoS("Finished syncing TrafficControl", "TrafficControl", tcName, "durationTime", time.Since(startTime))
+	}()
+
+	tc, err := c.trafficControlLister.Get(tcName)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			eState, exist := c.getTcState(tcName)
+			if !exist {
+				return nil
+			}
+			if err := c.uninstallTrafficControl(tcName, eState); err != nil {
+				return err
+			}
+			return nil
+		}
+		return err
+	}
+
+	tcState, exist := c.getTcState(tcName)
+
+	if !exist {
+		tcState = c.createTcState(tcName)
+		var dstInPort, dstOutPort uint32
+		if err := c.listOVSPorts(); err != nil {
+			return err
+		}
+
+		if dstOutPort, err = c.getDevicePort(tc.Spec.ReturnPort); err != nil && tc.Spec.Action != v1alpha2.ActionMirror {
+			return err
+		}
+		if dstInPort, err = c.getDevicePort(&tc.Spec.TargetPort); err != nil {
+			return err
+		}
+		if err := c.ofClient.InstallTrafficControlInitFlows(dstInPort, dstOutPort, tc.Spec.Action); err != nil {
+			klog.ErrorS(err, "Install the init flow failed", "Name", tcName)
+			return err
+		}
+		tcState.dstOfInPort = dstInPort
+		tcState.dstOfOutPort = dstOutPort
+	}
+	var pods []*v1.Pod
+	if pods, err = c.filterPods(&tc.Spec.AppliedTo); err != nil {
+		klog.ErrorS(err, "List Pods error", "Name", tcName, "selector", tc.Spec.AppliedTo)
+		return err
+	}
+	klog.InfoS("Filter Pods", "localPodsNum", len(pods), "selector", tc.Spec.AppliedTo)
+	if pods == nil {
+		return nil
+	}
+
+	var podOfPorts []uint32
+	for _, pod := range pods {
+		podItem := strings.Join([]string{pod.Namespace, pod.Name}, "/")
+		tcState.pods.Insert(podItem)
+
+		podInterfaces := c.interfaceStore.GetContainerInterfacesByPod(pod.Name, pod.Namespace)
+		if len(podInterfaces) == 0 {
```

Comment:
We should log when the interface is not found, otherwise we would never know something is wrong.